src/detection_win.cpp

```diff
@@ -468,79 +471,77 @@ void UpdateDevice(PDEV_BROADCAST_DEVICEINTERFACE pDevInf, WPARAM wParam, DeviceS
-		DWORD nSize=0 ;
-		TCHAR buf[MAX_PATH];
-
-		if (!DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, sizeof(buf), &nSize)) {
-			break;
-		}
-		NormalizeSlashes(buf);
-
-		if(szDevId == buf) {
-			WaitForSingleObject(deviceChangedSentEvent, INFINITE);
-
-			DWORD DataT;
-			DWORD nSize;
-			DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_LOCATION_INFORMATION, &DataT, (PBYTE)buf, MAX_PATH, &nSize);
-  		        DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)(buf + nSize - 1), MAX_PATH - nSize, &nSize);
-
-			if(state == DeviceState_Connect) {
-				DeviceItem_t* device = new DeviceItem_t();
-
-				AddItemToList(buf, device);
-				ExtractDeviceInfo(hDevInfo, pspDevInfoData, buf, MAX_PATH, &device->deviceParams);
+	if (pspDevInfoData) {
+		pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
+		for (int i = 0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
+			DWORD nSize = 0;
+			TCHAR buf[MAX_PATH];
 
-				currentDevice = &device->deviceParams;
-				isAdded = true;
+			if (!DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, sizeof(buf), &nSize)) {
+				break;
 			}
-			else {
-
-				ListResultItem_t* item = NULL;
-				if(IsItemAlreadyStored(buf)) {
-					DeviceItem_t* deviceItem = GetItemFromList(buf);
-					if(deviceItem)
-					{
-						item = CopyElement(&deviceItem->deviceParams);
+			NormalizeSlashes(buf);
+
+			if (szDevId == buf) {
+				WaitForSingleObject(deviceChangedSentEvent, INFINITE);
+
+				DWORD DataT;
+				DWORD nSize;
+				DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_LOCATION_INFORMATION, &DataT, (PBYTE) buf, MAX_PATH, &nSize);
+				DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)(buf + nSize - 1), MAX_PATH - nSize, &nSize);
+
+				if (state == DeviceState_Connect) {
+					DeviceItem_t *device = new DeviceItem_t();
+
+					AddItemToList(buf, device);
+					ExtractDeviceInfo(hDevInfo, pspDevInfoData, buf, MAX_PATH, &device->deviceParams);
+
+					currentDevice = &device->deviceParams;
+					isAdded = true;
+				} else {
+
+					ListResultItem_t *item = NULL;
+					if (IsItemAlreadyStored(buf)) {
+						DeviceItem_t *deviceItem = GetItemFromList(buf);
+						if (deviceItem) {
+							item = CopyElement(&deviceItem->deviceParams);
+						}
+						RemoveItemFromList(deviceItem);
+						delete deviceItem;
 					}
-					RemoveItemFromList(deviceItem);
-					delete deviceItem;
-				}
 
-				if(item == NULL) {
-					item = new ListResultItem_t();
-					ExtractDeviceInfo(hDevInfo, pspDevInfoData, buf, MAX_PATH, item);
+					if (item == NULL) {
+						item = new ListResultItem_t();
+						ExtractDeviceInfo(hDevInfo, pspDevInfoData, buf, MAX_PATH, item);
+					}
+					currentDevice = item;
+					isAdded = false;
 				}
-				currentDevice = item;
-				isAdded = false;
-			}
 
-			break;
+				break;
+			}
 		}
-	}
 
-	if (pspDevInfoData) {
```

Comment:
Why is this check removed but we are adding if statements around other areas?

Reply:
Previously, they first used a pointer, then checked for null before release. Now, first, check for null, then use and release

Reply:
I use CLion IDE, which has formatted the source code, so there are a lot of differences.